internal/migrations/migrator.go

```diff
@@ -54,30 +108,76 @@ type activeMigration struct {
 	ChainID string
 }
 
-// TODO: when we implement the constructor, we need to add a note that the snapshotter should
-// not be the normal snapshotter, but instead its own instance. This is needed to ensure we do not
-// delete the migration snapshot.
+func SetupMigrator(ctx context.Context, db Database, snapshotter Snapshotter, accounts SpendTracker, dir string, logger log.Logger) (*Migrator, error) {
+	// Set the migrator declared in migrations.go
+	migrator.snapshotter = snapshotter
```

Comment:
For future safety, can you return an error here if this is called more than once? Since this is used to initialize the singleton, _but_ it is also returned like a constructor, we should make sure people don't treat it as a constructor that can be called many times